### PR TITLE
[chore]: enable bool-compare rule from testifylint

### PR DESCRIPTION
**Pull request by mmorel-35**
#### Description

Testifylint is a linter that provides best practices with the use of testify.

This PR enables [bool-compare](https://github.com/Antonboom/testifylint?tab=readme-ov-file#bool-compare) rule from [testifylint](https://github.com/Antonboom/testifylint)

It's linter provided by golangci-lint.

Here all available rules are activated except those who require to be fixed. This PR only fixes bool-compare so the quantity of changes stays reasonnable for reviewers.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -199,6 +202,14 @@ golangci-lint/%: $(GOLANGCI_LINT)
 		&& cd $(DIR) \
 		&& $(GOLANGCI_LINT) run --allow-serial-runners $(ARGS)
 
+.PHONY: testifylint
+testifylint: $(OTEL_GO_MOD_DIRS:%=testifylint/%)
```
Comment (dmathieu) [inline Makefile:206]: Why do we need testifylint as a standalone command? We already have it as part of golangci-lint.

**Review by dmathieu — COMMENTED** [review]
This is nice, thank you!

File: Makefile (hunk 1 above)
Comment (dmathieu) [reply Makefile:206]: Can't you install it locally, without the makefile changes to be able to run it?

Comment (dmathieu) [reply Makefile:206]: You could also make the makefile changes locally, but not commit them.

Comment (dmathieu) [reply Makefile:206]: That doesn't seem so tedious:

```
git add . && git reset Makefile
```

Comment (pellared) [reply Makefile:206]: AFAIK `golangci-lint --fix` still returns an error if an issue is not fixed

**Review by dmathieu — APPROVED** [review]

**Review by pellared — APPROVED** [review]

**Review by MrAlias — APPROVED** [review]

**Pull request merged**